### DISCUSSION TITLE
ESP32 takes 22+ seconds to go do deepsleep with static IP

### DISCUSSION THREAD

**Discussion by ss7799166340** [arendst/Tasmota#16116] [Q&A] [answered]
I am using deepsleep with tasmota on ESP32, the problem seems to be too much of wake time, it takes 22 seconds  to wake up and connect to WIFI and send one MQTT message and then go back to sleep. I have also set the static IP. I am using the latest version of the firmware. any help to reduce the wake time would be very helpful, 22 seconds seems to be long and it effects the battery backup drastically.

**Comment by stefanbode** [comment, marked as answer]
Hi, the deepsleep was developed for the esp8266. The esp32 has never been optimized for it. One additional thing is that esp32 has much more options on deepsleep how and when to wakeup. Can You upload some detailed log files to get an idea where the time is lost.

**Reply by ss7799166340** [reply]
where can i find the activity log for tasmota, i mean we can see the console log but i dont think that shows what the tasmota is doing internally or does it ?

**Reply by sfromis** [reply]
You can get more details about some internal operations by setting log level 4 or 3. And just having the log including timestamps already tells a lot about where the time is "lost", even without having extra details.
